src/transformers/cache_utils.py

```diff
@@ -1660,7 +1660,10 @@ def get_max_length(self) -> Optional[int]:
         return self.max_cache_len
 
     def get_seq_length(self, layer_idx: Optional[int] = 0):
-        return None
+        # Occupied cache == any slot in the 3rd dim (sequence length) holds a non-zero value. To save on compute, let's
+        # limit the check to the first batch member and head dimension.
+        # TODO: deprecate this function in favor of `cache_position`
+        return (self.key_cache[layer_idx][0, 0].any(dim=-1)).sum()
```

Comment:
`HybridCache` is a `StaticCache` with alternating sliding window layers. The method to retrieve the cache length is copy/paste from `StaticCache`

We will want to use another method in the future, but let's leave this as a copy of `StaticCache` for now. This method is needed in the updated gemma 2.